### PR TITLE
add support for throwOnWarn option

### LINKED ISSUE #45: Add support for setting `throwOnWarn` option
After going through our what we're doing in our [`postProcess` hook](https://github.com/ember-cli/broccoli-lint-eslint/blob/master/lib/index.js#L182) -- checking for a `throwOnError` option and then reducing the severity count from a report result's `messages` [to a truthy number if we detect an error](https://github.com/ember-cli/broccoli-lint-eslint/blob/master/lib/index.js#L27) --  it seems like it would be pretty trivial to also check for `throwOnWarn` (or any arbitrary threshold, but perhaps that's another discussion 😄), reduce message severity into a raw sum, and then act accordingly from there. 

(Side note: We're actually iterating through messages before checking whether or not the user has even set `throwOnError`, but #44 addresses that as well)

### PR DESCRIPTION
**Pull request by BrianSipple**
Closes #45.
 
Mainly, this update is concerned with offering the ability to specify a `throwOnWarn` option, but it has the serendipitous effect of cleaning up or logic for "throwing" in general. `getResultsSeverity` now [computes a more honest value](https://github.com/ember-cli/broccoli-lint-eslint/pull/60/files#diff-6d186b954a58d5bb740f73d84fe39073R32) (instead of adding 1 for severity of 2 :smile:), and the [processString function](https://github.com/ember-cli/broccoli-lint-eslint/pull/60/files#diff-6d186b954a58d5bb740f73d84fe39073R216) can than use this value to make a more informed decision about what it wants to do.

One more thing to note, tough: I also removed the test for using a custom-rule.

We were getting lucky before in that they were being run last, but whenever that wasn’t the case, I’d keep running into the same issue that @nickiaconis [caught a while back](https://github.com/eslint/eslint/issues/5577) with the CLIEngine not necessarily removing the definition on the next test (and thus erring out on a `custom-rule` not being defined). 

Since custom rules are really just one more feature of ESLint itself (and are [well-tested there](https://github.com/eslint/eslint/blob/e13d87e0cc1626c29a73baa11f51d28b04e5741c/tests/fixtures/rules/eslint.json)), I feel like it's one more thing that should be removed from our own tests to make them cleaner.

### CODE REVIEW

**Comment by BrianSipple** [comment]
@Turbo87 Once this is squared away, I'm thinking we can also cut a 2.7.0 release and merge 2.x into master.

**Comment by Turbo87** [comment]
> One more thing to note, tough: I also removed the test for using a custom-rule.

I noticed the same issue. I'm not sure if removing the test is the best option, but I don't have a better solution for now...

File: lib/index.js [hunk 1]
```diff
@@ -9,20 +9,27 @@ const escapeStringRegexp = require('escape-string-regexp');
...
+ * @returns {Number} accumulatedSeverity The total severity from of the list of results
+ *    0 indicates all-clear
+ *    1 indicates a warning-level result
+ *    > 1 indicates an error-level result
  */
 function getResultSeverity(resultMessages = []) {
+  let severity;
+
   return resultMessages.reduce((accumulatedSeverity, message) => {
+    severity = message.severity || 0;
```
Comment (Turbo87) [inline lib/index.js]: why did you move the `let severity` out of the lambda? this often indicates that the function has side-effects but I don't see any in this case.

**Comment by Turbo87** [comment]
LGTM!

File: lib/index.js (hunk 1 above)
Comment (BrianSipple) [inline lib/index.js]: Good call 👍
